Index: vdr-1.7.29/config.c
===================================================================
--- vdr-1.7.29.orig/config.c	2012-07-19 00:58:14.516959336 +0200
+++ vdr-1.7.29/config.c	2012-07-19 02:28:38.453986006 +0200
@@ -468,6 +468,8 @@
   ChannelsWrap = 0;
   ShowChannelNamesWithSource = 0;
   EmergencyExit = 1;
+  JumpSeconds = 60;
+  JumpSecondsSlow = 10;
 }
 
 cSetup& cSetup::operator= (const cSetup &s)
@@ -668,6 +670,8 @@
   else if (!strcasecmp(Name, "ChannelsWrap"))        ChannelsWrap       = atoi(Value);
   else if (!strcasecmp(Name, "ShowChannelNamesWithSource")) ShowChannelNamesWithSource = atoi(Value);
   else if (!strcasecmp(Name, "EmergencyExit"))       EmergencyExit      = atoi(Value);
+  else if (!strcasecmp(Name, "JumpSeconds"))         JumpSeconds        = atoi(Value);
+  else if (!strcasecmp(Name, "JumpSecondsSlow"))     JumpSecondsSlow    = atoi(Value);
   else
      return false;
   return true;
@@ -771,6 +775,8 @@
   Store("ChannelsWrap",       ChannelsWrap);
   Store("ShowChannelNamesWithSource", ShowChannelNamesWithSource);
   Store("EmergencyExit",      EmergencyExit);
+  Store("JumpSeconds",        JumpSeconds);
+  Store("JumpSecondsSlow",    JumpSecondsSlow);
 
   Sort();
 
Index: vdr-1.7.29/config.h
===================================================================
--- vdr-1.7.29.orig/config.h	2012-07-19 00:58:14.576959339 +0200
+++ vdr-1.7.29/config.h	2012-07-19 02:28:38.453986006 +0200
@@ -338,6 +338,8 @@
   int ChannelsWrap;
   int ShowChannelNamesWithSource;
   int EmergencyExit;
+  int JumpSeconds;
+  int JumpSecondsSlow;
   int __EndData__;
   cString InitialChannel;
   cString DeviceBondings;
Index: vdr-1.7.29/menu.c
===================================================================
--- vdr-1.7.29.orig/menu.c	2012-07-19 02:28:38.421986003 +0200
+++ vdr-1.7.29/menu.c	2012-07-19 02:28:38.457986005 +0200
@@ -3511,6 +3511,8 @@
   Add(new cMenuEditBoolItem(tr("Setup.Replay$Jump&Play"), &data.JumpPlay));
   Add(new cMenuEditBoolItem(tr("Setup.Replay$Play&Jump"), &data.PlayJump));
   Add(new cMenuEditBoolItem(tr("Setup.Replay$Pause at last mark"), &data.PauseLastMark));
+  Add(new cMenuEditIntItem(tr("Setup.Replay$Jump Seconds"), &data.JumpSeconds));
+  Add(new cMenuEditIntItem(tr("Setup.Replay$Jump Seconds Slow"), &data.JumpSecondsSlow));
 }
 
 void cMenuSetupReplay::Store(void)
@@ -5451,9 +5453,9 @@
     case kRight:   Forward(); break;
     case kRed:     TimeSearch(); break;
     case kGreen|k_Repeat:
-    case kGreen:   SkipSeconds(-60); break;
+    case kGreen:   SkipSeconds(-Setup.JumpSeconds); break;
     case kYellow|k_Repeat:
-    case kYellow:  SkipSeconds( 60); break;
+    case kYellow:  SkipSeconds(Setup.JumpSeconds); break;
     case kPrev|k_Repeat:
     case kPrev:    if (lastSkipTimeout.TimedOut()) {
                       lastSkipSeconds = REPLAYCONTROLSKIPSECONDS;
@@ -5500,7 +5502,7 @@
                                   marks.Save();
                                   }
                                else
-                                  SkipSeconds(-20);
+                                  SkipSeconds(-Setup.JumpSecondsSlow);
                                break;
         case k3|k_Repeat:
         case k3:               if (GetReplayMode(play, forward, speed) && !play && GetIndex(Current, Total) && (m = marks.Get(Current)) != NULL) {
@@ -5513,7 +5515,7 @@
                                   marks.Save();
                                   }
                                else
-                                  SkipSeconds(20);
+                                  SkipSeconds(Setup.JumpSecondsSlow);
                                break;
         case kMarkToggle:      if (GetReplayMode(play, forward, speed) && play) {
                                   displayFrames = true;
